mobile/lib/providers/app_life_cycle.provider.dart

```diff
@@ -46,32 +46,40 @@ class AppLifeCycleNotifier extends StateNotifier<AppLifeCycleEnum> {
 
     // Needs to be logged in
     if (isAuthenticated) {
+      // switch endpoint if needed
+      await _ref.read(authProvider.notifier).setOpenApiServiceEndpoint();
+
       final permission = _ref.watch(galleryPermissionNotifier);
       if (permission.isGranted || permission.isLimited) {
-        _ref.read(backupProvider.notifier).resumeBackup();
-        _ref.read(backgroundServiceProvider).resumeServiceIfEnabled();
+        await _ref.read(backupProvider.notifier).resumeBackup();
+        await _ref.read(backgroundServiceProvider).resumeServiceIfEnabled();
       }
-      _ref.read(serverInfoProvider.notifier).getServerVersion();
+
+      await _ref.read(serverInfoProvider.notifier).getServerVersion();
+
       switch (_ref.read(tabProvider)) {
         case TabEnum.home:
-          _ref.read(assetProvider.notifier).getAllAsset();
+          await _ref.read(assetProvider.notifier).getAllAsset();
         case TabEnum.search:
         // nothing to do
         case TabEnum.albums:
-          _ref.read(albumProvider.notifier).refreshRemoteAlbums();
+          await _ref.read(albumProvider.notifier).refreshRemoteAlbums();
         case TabEnum.library:
         // nothing to do
       }
```

Comment:
Is it intentional that there are no break statements? e.g. `search` will also call `refreshRemoteAlbums`.

Reply:
It is good to put `break` in, from my testing, it doesn't continue down the cases if you are in `TabEnum.home`